wrappers/wrapper.rb

```diff
@@ -501,11 +501,29 @@ def build_rest(rest, vehicle = nil)
       }
     end
 
+    def build_skills(job)
+      return nil unless job
+
+      all_skills = job.skills - job.original_skills
+      skills_to_output = []
+
+      vehicle_cluster = all_skills.find{ |sk| sk.to_s.include?('vehicle_partition_') }
+      skills_to_output << vehicle_cluster.to_s.split('_')[2..-1].join('_') if vehicle_cluster
+
+      work_day_cluster = all_skills.find{ |sk| sk.to_s.include?('work_day_partition_') }
+      skills_to_output << work_day_cluster.to_s.split('_')[3..-1].join('_') if work_day_cluster
+
+      skills_to_output << job.original_skills
+
+      skills_to_output.flatten
+    end
+
     def build_detail(job, activity, point, day_index, job_load, vehicle, delivery = nil)
       {
         lat: point&.location&.lat,
         lon: point&.location&.lon,
-        skills: job&.skills,
+        skills: build_skills(job),
```

Comment:
can't we just call the `original_skills` ?

Reply:
No because we want to return skills corresponding to clusters.

Reply:
Oh in that case why don't we return the skills with the internal masking? Don't we want to make sure that the cluster skills are separate from non-clustering skills at the output.

If we remove the maskings `work_day_partition_X` and return `X`, if the user has skill `X` then it will clash with that. I don't think we have a verification for that. Then the user cannot know which is skill is due to clustering and which skill is due to their own skills.

Reply:
I am only returning _X_ because that was one of our user's need, to have the exact id of corresponding vehicle.

It seems like we wanted to make it easier, returning only _skills_ but we are making it harder. We should have _clustering_skills_ for every service and shipment, as we initially said (and add a new column with vehicle_cluster skills and day_cluster skills specifically in the CSV. 

@senhalil @braktar What do you think ?  Are we okay to increase number of attributes within service/shipment model ? I think it would simplify the code. But then we discussed about using internal skills for sticky_vehicles and that is why we decided to make everythink in _skills_.

Reply:
https://gitlab.com/mapotempo/optimizer-api/-/issues/697

Reply:
By the way do we know if they will not be okay with  `vehicle_cluster_vehicleid`? Maybe we are making it harder for no reason. These partition skills are internally generated so it is quite normal that they would have internal masks. 

If we decide to have separate columns in the CSV, why do we need clustering skills for the internal code? We can keep the current code (and not remove the skill mask for cluster skills) and CSV parser can parse `vehicle_cluster_X` into a separate column :thinking:

Reply:
We do know they can not work with _ vehicle_cluster_vehicleid_, it was like that and they still had troubles with it :/

We first did not want to return specific column for clustering but I guess that would not hurt. It is strange to mix internal and original skills.